examples/bri-3/src/bri/identity/bpiSubjects/agents/bpiSubjects.agent.ts

```diff
@@ -1,22 +1,52 @@
-import { BadRequestException, Injectable } from "@nestjs/common";
+import { BadRequestException, Injectable, NotFoundException } from "@nestjs/common";
 import { BpiSubject } from "../models/bpiSubject";
 import { BpiSubjectType } from "../models/bpiSubjectType.enum";
 
 import { uuid } from 'uuidv4'; 
+import { BpiSubjectStorageAgent } from "../persistence/bpiSubjectsStorage.agent";
+import { NotFoundError } from "rxjs";
  
+// Agent methods have extremely declarative names and perform a single task 
 @Injectable()
 export class BpiSubjectAgent {
-  // Agent methods have extremely declarative names and perform a single task 
+
+  constructor(private repo: BpiSubjectStorageAgent) { }
   public throwIfCreateBpiSubjectInputInvalid(name: string, desc: string, pk: string) {
     // This is just an example, these fields will be validated on the DTO validation layer
     // This validation would check internal business rules (i.e. bpiSubject must have public key in the format defined by the participants..) 
     if (!name) {
-      // We stop execution in case of critical errors by throwing a simple exception
       throw new BadRequestException("Name cannot be empty.")
     };
   }
 
   public createNewExternalBpiSubject(name :string, description: string, publicKey: string): BpiSubject {
     return new BpiSubject(uuid(), name, description, BpiSubjectType.External, publicKey );
   }
+
+  public async fetchUpdateCandidateAndThrowIfUpdateValidationFails(id: string, name: string, desc: string, pk: string): Promise<BpiSubject> {
+    const bpiSubjectToUpdate = await this.repo.getBpiSubjectById(id);
+
+    if(!bpiSubjectToUpdate) {
+      throw new NotFoundException(`Bpi Subject with id: ${id} does not exist.`)
```

Comment:
ok a nitpick but leading to a more general point -> should we not have an error/notification message file with the messages in it in case we need to use them more than once? cc @skosito @Kasshern @Manik-Jain @Ybittan

Reply:
Yes, good point. I ll introduce constants for now and we can later think of internalization.

Reply:
Added a primitive mechanism for shared err messages. Let's build on it further as we go.